### PR TITLE
Update code-scanning.md

### PR DESCRIPTION
**Pull request by cmboling**
Stretch exercises on how to upload a SARIF and CodeQL databases as workflow artifacts

### CODE REVIEW

File: exercises/code-scanning.md [hunk 1]
```diff
@@ -318,5 +318,89 @@ queries:
...
+
+<details>
+<summary>Solutions</summary>
+
+```yaml 
+    - name: Upload CodeQL database
+      id: codeql-database-bundle
+      env:
+        LANGUAGE: ${{ matrix.language }}
+        CODEQL_PATH: /opt/hostedtoolcache/CodeQL/<codeql-bundle-name>/x64/codeql/codeql
```
Comment (knewbury01) [inline exercises/code-scanning.md:361]: I would maybe mention how to explicitly find the `<codeql-bundle-name>` part of the path
and/or highlight that is an option that needs to be filled in for the solution.

Unless its actually super obvi from the [doc](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#tools) you listed above, but the way I found its value was in the sarif artifact I could see `/opt/hostedtoolcache/CodeQL/0.0.0-20210622/x64/codeql/qlpacks/codeql-go-examples/` which means the bundle name being used is `0.0.0-20210622` in my runner, and filling in that value enables this step to work, but finding that was... maybe not totally intuitive. Also possible I took the convoluted route though!

File: exercises/code-scanning.md [hunk 2]
```diff
@@ -318,5 +318,89 @@ queries:
 ```
 </details>
 
+#### _Stretch Exercise 4a: Uploading the SARIF as a workflow artifact_
```
Comment (knewbury01) [inline exercises/code-scanning.md:321]: love this! super cool ! 👍 

maybe we could add one line describing why we might want to obtain the sarif artifact, other than having it as the complete record of results?

File: exercises/code-scanning.md [hunk 3]
```diff
@@ -318,5 +318,89 @@ queries:
...
+      uses: actions/upload-artifact@v2
+      with:
+        name: sarif
+        path: code-scanning-results
+        retention-days: 7
+```
+</details>
+    
+#### _Stretch Exercise 4b: Uploading CodeQL databases as workflow artifacts_
+
```
Comment (knewbury01) [inline exercises/code-scanning.md:343]: also very cool addition! 👍

File: exercises/code-scanning.md [hunk 4]
```diff
@@ -318,5 +318,89 @@ queries:
 ```
 </details>
 
+#### _Stretch Exercise 4a: Uploading the SARIF as a workflow artifact_
+    
+The output of the `github/codeql-action/analyze@v1` is a SARIF. What action should we use to upload the SARIF as an artifact?
```
Comment (cmboling) [inline exercises/code-scanning.md]:
```suggestion
The output of the `github/codeql-action/analyze@v1` is a SARIF. You may want to obtain this when you want to look into the SARIF directly on your local machine and/or view it in SARIF viewer tool outside of GitHub. What action should we use to upload the SARIF as an artifact?
```

File: exercises/code-scanning.md [hunk 5]
```diff
@@ -318,5 +318,89 @@ queries:
...
+      with:
+        name: sarif
+        path: code-scanning-results
+        retention-days: 7
+```
+</details>
+    
+#### _Stretch Exercise 4b: Uploading CodeQL databases as workflow artifacts_
+    
+By looking at the logs, where does CodeQL output the CodeQL databases, and similar to the previous exercise, how do we upload this?
```
Comment (cmboling) [inline exercises/code-scanning.md]:
```suggestion
By looking at the logs, where does CodeQL output the CodeQL databases, and similar to the previous exercise, how do we upload this? Furthermore, you'll be able to tell where the CodeQL binary lives as well, so you can pull the path to the CodeQL binary on the GitHub hosted runner into the Actions workflow.
```

**Comment by cmboling** [comment]
@knewbury01 thanks for the suggestions! I'm merging this now because a couple of people are interested in the additional steps I've defined and this is the best place to show those steps 😄

**Pull request merged**